engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,394 @@
+                final int len) {
+            throw new UnsupportedOperationException("RollingFormula is not supported in cumulative operations.");
+        }
+
+        @Override
+        public void accumulateRolling(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] influencerValueChunkArr,
+                @Nullable final LongChunk<OrderedRowKeys> affectedPosChunk,
+                @Nullable final LongChunk<OrderedRowKeys> influencerPosChunk,
+                @NotNull final IntChunk<? extends Values> pushChunk,
+                @NotNull final IntChunk<? extends Values> popChunk,
+                final int len) {
+
+            // TODO: reuse a
+            setValueChunks(influencerValueChunkArr);
+            setPosChunks(affectedPosChunk, influencerPosChunk);
+
+            int pushIndex = 0;
+
+            // chunk processing
+            for (int ii = 0; ii < len; ii++) {
+                final int pushCount = pushChunk.get(ii);
+                final int popCount = popChunk.get(ii);
+
+                if (pushCount == NULL_INT) {
+                    outputValues.fillWithNullValue(ii, 1);
+                    continue;
+                }
+
+                // pop for this row
+                if (popCount > 0) {
+                    for (RingBufferWindowConsumer consumer : inputConsumers) {
+                        consumer.pop(popCount);
+                    }
+                }
+
+                // push for this row
+                if (pushCount > 0) {
+                    for (RingBufferWindowConsumer consumer : inputConsumers) {
+                        consumer.push(pushIndex, pushCount);
+                    }
+                    pushIndex += pushCount;
+                }
+
+                // If not empty (even if completely full of null), run the formula over the window values.
+                outputSetter.accept(ii);
+            }
+
+            // chunk output to column
+            writeToOutputColumn(inputKeys);
+        }
+
+        @Override
+        protected void writeToOutputChunk(int outIdx) {
+            throw Assert.statementNeverExecuted("RollingFormulaMultiColumnOperator.Context.writeToOutputChunk");
+        }
+
+        @Override
+        public void writeToOutputColumn(@NotNull final RowSequence inputKeys) {
+            primitiveOutputSource.fillFromChunk(outputFillContext, outputValues, inputKeys);
+        }
+
+        @Override
+        public void reset() {
+            // Clear all the ring buffers for re-use
+            for (RingBufferWindowConsumer consumer : inputConsumers) {
+                consumer.reset();
+            }
+            nullCount = 0;
+        }
+
+        @Override
+        public void close() {
+            outputValues.close();
+            outputFillContext.close();
+        }
+    }
+
+    public RollingFormulaMultiColumnOperator(
+            @NotNull final MatchPair pair,
```

Comment:
Please added Javadoc for this method, especially what it means when the nullable column is actually null.